apps/game_backend/lib/game_backend/matches.ex

```diff
@@ -0,0 +1,22 @@
+defmodule GameBackend.Matches do
+  @moduledoc """
+  Matches
+  """
+  alias Ecto.Multi
+  alias GameBackend.Repo
+  alias GameBackend.Matches.ArenaMatchResult
+
+  def create_arena_match_results(match_id, results) do
+    transaction = Multi.new()
+
+    {transaction, _} =
+      Enum.reduce(results, {transaction, 1}, fn result, {transaction_acc, idx} ->
+        attrs = Map.put(result, "match_id", match_id)
+        changeset = ArenaMatchResult.changeset(%ArenaMatchResult{}, attrs)
+        transaction_acc = Multi.insert(transaction_acc, {:insert, idx}, changeset)
+        {transaction_acc, idx + 1}
+      end)
```

Comment:
i think you can omit this indexing by using the client id in the transaction key:

```
 Enum.reduce(results, Multi.new(), fn result, transaction_acc ->
        attrs = Map.put(result, "match_id", match_id)
        changeset = ArenaMatchResult.changeset(%ArenaMatchResult{}, attrs)
        Multi.insert(transaction_acc, {:insert, attrs.user_id}, changeset)
      end)
```

Reply:
also, if we're going to include the match id in the registry why send them in a separate field? maybe we could include the match_id when we create the attrs map in arena an match the whole map in `match_results_controller`

Reply:
Changed [622cd6d](https://github.com/lambdaclass/mirra_backend/pull/529/commits/622cd6d101e20b92d86fde749ce12600b3598d30)